beluga/include/beluga/algorithm/spatial_hash.hpp

```diff
@@ -196,6 +196,61 @@ class spatial_hash<Sophus::SE2d, void> {
   spatial_hash<std::tuple<double, double, double>> underlying_hasher_{{1., 1., 1.}};
 };
 
+/**
+ * Specialization for Sophus::SE3d. Will calculate the spatial hash based on the translation and rotation (expressed in
+ * Euler Angles).
+ */
+template <>
+class spatial_hash<Sophus::SE3d, void> {
+ public:
+  /// Constructs a spatial hasher given per-coordinate resolutions.
+  /**
+   * \param x_clustering_resolution Clustering resolution for the X axis, in meters.
+   * \param y_clustering_resolution Clustering resolution for the Y axis, in meters.
+   * \param z_clustering_resolution Clustering resolution for the Z axis, in meters.
+   * \param angular_clustering_resolution Clustering resolution for roll pitch and yaw, in radians.
+   */
+  explicit spatial_hash(
+      double x_clustering_resolution,
+      double y_clustering_resolution,
+      double z_clustering_resolution,
+      double angular_clustering_resolution)
```

Comment:
@serraramiro1 meta: why the same resolution for all rotation axis?

Reply:
Convenience. Added a new overload with different resolution for every element